compiler/cpp/src/parser/ast/ComBlockExprAST.cpp

```diff
@@ -1,7 +1,24 @@
 // Copyright (c) 2021 Compose Generator Contributors. All rights reserved.
 
 #include "ComBlockExprAST.h"
+#include "ComLineBlockExprAST.h" // This import cant be in the header file for some reason
+#include "ComBlockBlockExprAST.h" // This import cant be in the header file for some reason
 
 ComBlockExprType ComBlockExprAST::getType() {
     return type;
+}
+
+std::string ComBlockExprAST::serialize() {
+    switch (type) {
+        case COM_LINE_BLOCK_EXPR: {
+            auto* comLineBlock = static_cast<ComLineBlockExprAST*>(this);
+            return comLineBlock->serialize();
+        }
+        case COM_BLOCK_BLOCK_EXPR: {
+            auto* comBlockBlock = static_cast<ComBlockBlockExprAST*>(this);
+            return comBlockBlock->serialize();
+        }
+        default:
+            throw std::runtime_error("Unknown com block type");
```

Comment:
Shouldn't there be no `ComBlock ExprAST` at all if the type is not correct? But still, good practice to throw an error here.